### PR TITLE
WIP: Added support for configuration file parsing.

### LINKED ISSUE #35: Persistent configuration
In this PR https://github.com/it-is-wednesday/cute-sway-recorder/issues/33#issuecomment-2094851170 there was mention of setting defaults in a config file in `~/.config/cute-sway-recorder/defaults.{ext}`
It was mentioned that it should use [`configparser`](https://docs.python.org/3.7/library/configparser.html) or [`tomllib`](https://docs.python.org/3.11/library/tomllib.html) (which only supports >3.11)

### PR DESCRIPTION
**Pull request by ketexon**
Hello :D!!! I was working on #35 and have a draft. 

I created a new dataclass for the parsed config (called `ConfigFile`). It includes everything in `Config` except `selection` (I wasn't sure how I'd include it).

I tested `file_dest`s with/without `~` and made sure it doesn't accept directories, paths without names (`/`), or paths without an existing parent directory.
It also rejects negative/non-integer `delay` values and non-bool-like `include_audio` values.
When it rejects a value, it *should* print a descriptive error to console and continue running without throwing!

@kohane27 If you could help test I would greatly appreciate :3

closes #35

### CODE REVIEW

**Comment by it-is-wednesday** [comment]
Incredible job, I highly appreciate that you went through the issues to get a grip of the project's current state. Beautiful. And you code is neat as well! Looking forward to this one;)

File: cute_sway_recorder/config_file.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    # "foo/..".parent == "foo"
+    if not file_dest.expanduser().resolve().parent.is_dir():
+        raise ValueError(f"\"{file_dest}\"'s parent directory does not exist.")
+
+@dataclass
+class ConfigFile:
+    """
+    Contains data parsed from the config file 
+    at `~/.config/cute-sway-recording/defaults.conf`
+    Use :func:`ConfigFile.parse` to parse the ConfigFile.
```
Comment (it-is-wednesday) [inline cute_sway_recorder/config_file.py:32]: Damn, I didn't know that Sphinx notation works in docstrings. TIL, nice!

**Comment by it-is-wednesday** [comment]
I think `defaults.conf` is a bit confusing, because it implies that this file shouldn't be edited (at least that's how I interpret this name). 

I understand _why_ you chose this name, though – these are options that are passed by default to wf-recorder, and this reasoning is perfectly fine.

I suggest we use a neutral name, like "`~/.config/cute-sway-recorder/config.ini`", and inside this file we can create a `wf-recorder-defaults` table, under which every variable will be specified.

Also, do you mind throwing an example config file anywhere in the README? Or in the docstrings, or wherever you like.

Thank you very much friend :)

File: cute_sway_recorder/config_file.py (hunk 1 above)
Comment (ketexon) [reply cute_sway_recorder/config_file.py:32]: i didnt know it was called Sphinx notation :p TIL

**Comment by ketexon** [comment]
> I think `defaults.conf` is a bit confusing, because it implies that this file shouldn't be edited (at least that's how I interpret this name).
> 
> I understand _why_ you chose this name, though – these are options that are passed by default to wf-recorder, and this reasoning is perfectly fine.
> 
> I suggest we use a neutral name, like "`~/.config/cute-sway-recorder/config.ini`", and inside this file we can create a `wf-recorder-defaults` table, under which every variable will be specified.
> 
> Also, do you mind throwing an example config file anywhere in the README? Or in the docstrings, or wherever you like.
> 
> Thank you very much friend :)

Yes I think `config.ini` with `wf-recorder-defaults` sounds better. Using `DEFAULT` felt a little weird. Ill put an example in the readme too in a sec :3 thank u friend!!

**Comment by it-is-wednesday** [comment]
Beautiful!

**Comment by it-is-wednesday** [comment]
Lmao I didn't notice that it was actually ready when I commented "Beautiful" haha. Thanks for the patch!!

**Pull request merged**